README.md

```diff
@@ -50,3 +52,18 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+Time complexity: The randomroute function has a complexity of O(n) because the shuffling is linear, the distance iterates through all the cities in the route which is also O(n), in optiswap the slice and reverse takes O(n) and the while loop at worst case will run 100 consecutive times so the total complexity in O(n+n+n+100(n)) which simplifies to just O(n).
```

Comment:
How many times could you find an improving tour?

Reply:
Can you please clarify, are you asking how many times the loop could reset to 0?

Reply:
Yes. Your analysis is assuming that you never find an improving tour.